src/pyodide/internal/workers.py

```diff
@@ -0,0 +1,54 @@
+# This module defines a Workers API for Python. It is similar to the API provided by
+# JS Workers, but with changes and additions to be more idiomatic to the Python
+# programming language.
+from http import HTTPMethod, HTTPStatus
+
+import js
+
+from pyodide.ffi import to_js
+from pyodide.http import FetchResponse, pyfetch
+
+JSBody = (
+    "js.Blob | js.ArrayBuffer | js.TypedArray | js.DataView | js.FormData |"
+    "js.ReadableStream | js.URLSearchParams"
+)
+Body = "str | JSBody"
+Headers = dict[str, str] | list[tuple[str, str]]
+
+
+async def fetch(
+    resource: str,
+    headers: Headers = None,
+    body: Body | None = None,
+    method: HTTPMethod = HTTPMethod.GET,
+    **other_options,
+) -> FetchResponse:
+    if headers:
+        other_options["headers"] = headers
+
+    if body:
+        other_options["body"] = body
```

Comment:
To give a type to a kwargs argument, you can make a typed dict:
```py
class FetchKwargs(TypedDict, total=False):
    headers: Headers
    body : Body
```
and then you can do:
```py
async def fetch(
    resource: str,
    **kwargs: Unpack[FetchKwargs]
)
```
 and then you can get rid of this stuff.

Reply:
Cool trick, thanks!